service/service.go

```diff
@@ -22,27 +22,31 @@ import (
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/multierr"
+	"go.uber.org/zap"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/service/internal"
 	"go.opentelemetry.io/collector/service/internal/extensions"
 	"go.opentelemetry.io/collector/service/internal/pipelines"
+	"go.opentelemetry.io/collector/service/internal/telemetry"
 	"go.opentelemetry.io/collector/service/internal/telemetrylogs"
 )
 
 // service represents the implementation of a component.Host.
 type service struct {
-	buildInfo component.BuildInfo
-	config    *Config
-	telemetry component.TelemetrySettings
-	host      *serviceHost
+	buildInfo    component.BuildInfo
+	config       *Config
+	telemetry    component.TelemetrySettings
+	host         *serviceHost
+	colTelemetry *colTelemetry
 }
 
 func newService(set *settings) (*service, error) {
 	srv := &service{
 		buildInfo: set.BuildInfo,
 		config:    set.Config,
 		telemetry: component.TelemetrySettings{
+			Logger:         zap.NewNop(),
 			TracerProvider: trace.NewNoopTracerProvider(),
```

Comment:
Why do we initialize with Nop instances here, and later on lines 62-70 immediately reassign the values? Looks like  this Nop initialization is not needed.

Reply:
Yes, https://github.com/open-telemetry/opentelemetry-collector/pull/5570 fixed it for TraceProvider. We have the same issue with the Logger here.